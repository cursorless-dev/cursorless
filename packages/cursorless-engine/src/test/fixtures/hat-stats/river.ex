defmodule MyApp.Mod do
  alias MyApp.Car
​
  @model_regex ~r/^[A-Z][a-Z]*$/
​
  @doc """
  This is a function.
​
  ## People use markdown in docstrings usually
    * So lists
    * And inline code for arguments like `color` is common
  """
  def database_call(color) do
    query =
      from(c in Car,
        where:
          c.make == :toyota or
            fragment("lower(?) like ?", s.color, ^"#{color}%")
      )
​
    results =
      query
      |> Repo.all()
      |> Enum.filter(fn
        %{color: :unknown} -> false
        c -> c.make == Application.fetch_env!(__MODULE__, :make_filter)
      end)
      |> Enum.map(&[&1.model, &1.color])
​
    Enum.at(results, 0).model =~ @model_regex
  end
end
​
defmodule MyApp.Car do
  @moduledoc """
  A database model
  """
  use Ecto.Schema
  import Ecto.Changeset
​
  schema "cars" do
    field(:color, Ecto.Enum,
      values: [:blue, :unknown, :green, :silver, :black, :yellow, :red, :white]
    )
​
    field(:model, :string)
    field(:make, :string)
​
    belongs_to(:owner, MyApp.Person)
​
    timestamps()
  end
​
  def changeset(sign, attrs) do
    required_fields = [
      :make,
      :model
    ]
​
    optional_fields = [
      :owner
    ]
​
    sign
    |> cast(attrs, required_fields ++ optional_fields)
    |> validate_required(required_fields)
    |> foreign_key_constraint(:people, name: :cars_owned_by_fkey)
  end
end
