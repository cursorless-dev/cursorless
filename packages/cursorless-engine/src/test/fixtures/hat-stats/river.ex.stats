nTokens: 344

tokenSpark:
	▁▁▁▁▃▁▁▃▁█▁▁▁▁▁▁▁▃▁▁▁▃▃▃▃▃▁▁▃█▃▃▁▁▁▃▁▃█▁▁▁▁▃▁▁▃▃▃▃▃▃▃▁▁▃▁▃▃▃
	▃▃▃▁▁▃▁▃▁▁▃▃▃▃▃▁▁▁▃▃▃▃▃▃▃▃▃▃▃▃▁▃▃▃▃▃▃▃▃▃▃▃▃▁▃▁▁▃▃▃█▃▃▃▁▁▃▃▃▃
	▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▁▁▃▁▃▃▃▃▃▃▃▃▃▃▃▃▁▃▃▁▃▃▃▃▃▃▃▃▃█▃▃▃▃▃
	▃▁▃▅▃▃▃▃▁▃▃█▃▃▅▃▃▃▃▃▃▃▃▃▃▃▅▁▃▃▅▃█▃▃▃▃▃▃▃▃▃▃▃▅▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃
	▃▃▃▃▃▃▃▅▅▃▅▅▃▃▃█▃▃▅▃▅▅▃▃▃▃▅▃▅▅▃▃█▃▃▅▃▅▃▅▃▃█▃▅▅▃█▃▃▅▃▅▃▅▃▃▃▃▅
	▃▅▅▃▃█▃▃▃▅▃▃█▃▃▃▃▅▃▅▃▁▃▅▃▃▃▅▃▅▃▃▃▅▅▃▅▃▅▅▃▅▃▃

nHats:
	mean: 96%
	std: 0%
	min: 96%
	max: 96%
	spark: 0              25               50               75           █   100

nPenalty0:
	mean: 16%
	std: 0%
	min: 16%
	max: 16%
	spark: 0        █     25               50               75               100

nPenalty1:
	mean: 69%
	std: 0%
	min: 68%
	max: 69%
	spark: 0              25               50          █    75               100

nPenalty2:
	mean: 11%
	std: 0%
	min: 11%
	max: 12%
	spark: 0      █▁      25               50               75               100
