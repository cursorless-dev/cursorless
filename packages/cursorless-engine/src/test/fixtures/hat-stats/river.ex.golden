_         _    _ _  A
defmodule MyApp.Mod do
[-------] [---]|[-] []

  _      _   A_
  alias MyApp.Car
  [---] [---]|[-]

​

  _   _        _____A___ABAAA__A
  @model_regex ~r/^[A-Z][a-Z]*$/
  |[---------] |||||||||||||||||

​

  AB   _
  @doc """
  |[-] [-]

  _    _  C   _     B
  This is a function.
  [--] [] | [------]|

​

  _  _      _      _      A    _           _
  ## People use markdown in docstrings usually
  [] [----] [-] [------] [] [--------] [-----]

    A A     A
    * So lists
    | [] [---]

    B  B   C     A    _     _         A  _B    A A       D
    * And inline code for arguments like `color` is common
    | [-] [----] [--] [-] [-------] [--] |[---]| [] [----]

  A
  """
  [-]

  C       _        _C    _ D
  def database_call(color) do
  [-] [-----------]|[---]| []

    _     _
    query =
    [---] |

      A   AD B  E  _
      from(c in Car,
      [--]|| [] [-]|

        _    _
        where:
        [---]|

          FCA    A  AB      A
          c.make == :toyota or
          ||[--] [] |[----] []

            B       BBA    C_A B    ACA BD1    B ADA_2    __EB
            fragment("lower(?) like ?", s.color, ^"#{color}%")
            [------]||[---]||| [--] ||| ||[---]| ||||[---]||||

      C
      )
      |

​

    A       B
    results =
    [-----] |

      A
      query
      [---]

      __ B   ED  DD
      |> Repo.all()
      || [--]|[-]||

      AA A   FC     ED
      |> Enum.filter(fn
      || [--]|[----]|[]

        AA3    B CA      A  B E
        %{color: :unknown} -> false
        ||[---]| |[-----]| [] [---]

        4  C 51B    C  E          2        __F_         C D    A      E
        c -> c.make == Application.fetch_env!(__MODULE__, :make_filter)
        | [] ||[--] [] [---------]|[-------]||[--------]| |[---------]|

      B  F
      end)
      [-]|

      BD C   3C  1_BA_4D    D BA56    B1
      |> Enum.map(&[&1.model, &1.color])
      || [--]|[-]||||||[---]| |||[---]||

​

                       A
    D   6F 2C      E _21E     DA B          _
    Enum.at(results, 0).model =~ @model_regex
    [--]|[]|[-----]| |||[---] || |[---------]

  E
  end
  [-]

F
end
[-]

​

               A
E         F    2 1  F
defmodule MyApp.Car do
[-------] [---]|[-] []

  C1         F
  @moduledoc """
  |[-------] [-]

  2     A    2
  A database model
  | [------] [---]

  1
  """
  [-]

          A
  B   1   3  _
  use Ecto.Schema
  [-] [--]|[----]

             A
  C      2   4 A
  import Ecto.Changeset
  [----] [--]|[-------]

​

  C      2 3  3 1
  schema "cars" do
  [----] |[--]| []

                      A
    F    3E B   F 3   54   1
    field(:color, Ecto.Enum,
    [---]||[---]| [--]|[--]|

                                                                A A   A A
      A     F C1B   2 2C      3 3A    4 4   B  5 5C    6 6A     1 1D  2 2A    C
      values: [:blue, :unknown, :green, :silver, :black, :yellow, :red, :white]
      [----]| ||[--]| |[-----]| |[---]| |[----]| |[---]| |[----]| |[-]| |[---]|

    3
    )
    |

​

          A     A A
    1    433    3 4D     4
    field(:model, :string)
    [---]||[---]| |[----]|

          A    A A
    2    554   4 6E     5
    field(:make, :string)
    [---]||[--]| |[----]|

​

               B     A      A
    D         61C    5 5    6A     6
    belongs_to(:owner, MyApp.Person)
    [--------]||[---]| [---]|[----]|

​

              AA
    C         11
    timestamps()
    [--------]||

  5
  end
  [-]

​

               A    A      A
  2    B       2F   6 4    2 3
  def changeset(sign, attrs) do
  [-] [-------]|[--]| [---]| []

      B             E D
    required_fields = [
    [-------------] | |

      B    B
      26   1
      :make,
      |[--]|

      B
      3 D
      :model
      |[---]

    D
    ]
    |

​

            B       F E
    optional_fields = [
    [-------------] | |

      B
      4E
      :owner
      |[---]

    E
    ]
    |

​

    1
    sign
    [--]

           A     B                                   A
    CE  5  36    2   C             _          C      3
    |> cast(attrs, required_fields ++ optional_fields)
    || [--]|[---]| [-------------] [] [-------------]|

                        A               A
    DF C                4  D            4
    |> validate_required(required_fields)
    || [---------------]|[-------------]|

                             AB      B     B C                  A
    E1        D              55B     3 E   6 1           E      5
    |> foreign_key_constraint(:people, name: :cars_owned_by_fkey)
    || [--------------------]||[----]| [--]| |[----------------]|

  6
  end
  [-]

 F
end
[-]
